chain/sync.go

```diff
@@ -854,7 +863,7 @@ func (syncer *Syncer) checkBlockMessages(ctx context.Context, b *types.FullBlock
 		}
 
 		if err := sigs.Verify(&m.Signature, kaddr, m.Message.Cid().Bytes()); err != nil {
-			return xerrors.Errorf("secpk message %s has invalid signature: %w", m.Cid(), err)
+			return validation.ErrorWrapString(validation.ErrInvalidSecpkSignature, fmt.Sprintf("%s: %s", err.Error(), m.Cid()))
```

Comment:
note that from xerrors.Errorf, we also get stack trace information. i think thats generally worth preserving

Reply:
Agreed, I'll add the `Frame`. This is a good example of the error decoupling I would like to achieve, the type of error which might be referenced many times should be independent of the generated trace of where it is used.